File: backend/app/schemas/quote.py
from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel, Field


class QuoteCreate(BaseModel):
    professional_service_id: int = Field(gt=0)

    amount: Decimal = Field(
        gt=0,
        max_digits=10,
        decimal_places=2,
    )

    note: str | None = None

    expires_at: datetime | None = None


class QuoteResponse(BaseModel):
    id: int
    service_request_id: int
    professional_service_id: int
    amount: Decimal
    note: str | None
    status: str
    expires_at: datetime | None
    created_at: datetime
    updated_at: datetime

    model_config = {
        "from_attributes": True,
    }


class AdminQuoteResponse(BaseModel):
    id: int
    service_request_id: int
    professional_service_id: int

    customer_profile_id: int
    customer_name: str

    professional_profile_id: int
    professional_name: str

    service_id: int
    service_name: str

    amount: Decimal
    note: str | None
    status: str
    expires_at: datetime | None
    created_at: datetime
    updated_at: datetime

File: frontend/src/types/quote.ts
/*
|--------------------------------------------------------------------------
| Quote Types
|--------------------------------------------------------------------------
|
| Matches the backend QuoteResponse schema:
|
| id
| service_request_id
| professional_service_id
| amount
| note
| status
| expires_at
| created_at
| updated_at
|
|--------------------------------------------------------------------------
*/

export interface QuoteResponse {
  id: number;
  service_request_id: number;
  professional_service_id: number;
  amount: string;
  note: string | null;
  status: string;
  expires_at: string | null;
  created_at: string;
  updated_at: string;
}